utils/pools.ts

```diff
@@ -205,3 +204,46 @@ export const buildDefaultNextPoolState = (pool: Pool): NextPoolState => {
         expectedFrontRunningOraclePrice: pool.oraclePrice,
     };
 };
+
+export const saveImportedPoolsToLocalStorage: (customPools: string[]) => void = (customPools) => {
+    const importedPools = localStorage.getItem('importedPools');
+    if (!importedPools) {
+        // create new localStorage variable to store imported pools
+        localStorage.setItem('importedPools', JSON.stringify(customPools));
+    } else {
+        const parsedImportedPools = JSON.parse(importedPools);
+        customPools.forEach((pool) => {
+            if (!parsedImportedPools.includes(pool)) {
+                parsedImportedPools.push(pool);
+            }
+        });
+        localStorage.setItem('importedPools', JSON.stringify(parsedImportedPools));
+    }
+};
+
+export const removeImportedPoolFromUi: (
+    network: KnownNetwork,
+    poolAddress: string,
+    removePool: (network: KnownNetwork, pool: string) => void,
+) => void = (network, poolAddress, removePool) => {
+    try {
+        if (poolAddress) {
+            // Check for imported Pool in localStorage
+            const localStoragePoolAddresses = localStorage.getItem('importedPools');
+            const parsedImportedPools = localStoragePoolAddresses && JSON.parse(localStoragePoolAddresses);
+            const poolIndex = parsedImportedPools?.findIndex((pool: string) => pool === poolAddress);
+            if (poolIndex !== -1) {
+                parsedImportedPools.splice(poolIndex, 1);
+                localStorage.setItem('importedPools', JSON.stringify(parsedImportedPools));
+            }
+
+            // Remove URL parameters without reloading
+            window.history.pushState({}, document.title, window.location.pathname);
+
+            // Remove imported Pool from store
+            removePool(network, poolAddress);
+        }
+    } catch (err) {
+        console.error('Failed to send Buy action to Segment', err);
```

Comment:
we should also reset `importedPools` in localStorage back to an empty value here in the case where the error stems from being unable to parse it